### PR TITLE
New fingerprint format and optionally non-blocking

### PR DESCRIPTION
**Pull request by moxie0**

### CODE REVIEW

**Comment by FeuRenard** [comment]
- the hint "tap to scan" is invisible with the dark theme (see screenshots in #5664)
- When I try to compile for Android 4.4.4 I get the following error

> Error:The number of method references in a .dex file cannot exceed 64K.
Learn how to resolve this issue at https://developer.android.com/tools/building/multidex.html

- When I try to open Signal on Android 6.0 I get the following error: [logcat](https://gist.github.com/FeuRenard/85499c1790f3aa34caf5963578585762). It crashes directly after I open it. State of the app is a fresh install, so Signal attempts to start with the registration.

**Comment by FeuRenard** [comment]
For anyone else, who wants to use another branch after trying this one:
You have to downgrade the gradle plugin version again in Android Studio. For me this worked best when I played around with the gradle "Project-level settings" and restarted Android Studio.

File -> Settings -> Build, Execution, Deployment -> Build Tools -> Gradle

File: src/org/thoughtcrime/securesms/VerifyIdentityActivity.java [hunk 1]
```diff
@@ -17,155 +16,379 @@
...
-    this.masterSecret = masterSecret;
     getSupportActionBar().setDisplayHomeAsUpEnabled(true);
-    getSupportActionBar().setTitle(R.string.AndroidManifest__verify_identity);
+    getSupportActionBar().setTitle(R.string.AndroidManifest__verify_safety_numbers);
+
+    Recipient recipient = RecipientFactory.getRecipientForId(this, getIntent().getLongExtra(RECIPIENT_ID, -1), true);
+
+    Bundle extras = new Bundle();
+    extras.putParcelable(VerifyDisplayFragment.REMOTE_IDENTITY, getIntent().getParcelableExtra(RECIPIENT_IDENTITY));
+    extras.putString(VerifyDisplayFragment.REMOTE_NUMBER, recipient.getNumber());
```
Comment (michaelkirk) [inline src/org/thoughtcrime/securesms/VerifyIdentityActivity.java]: I believe this needs to be canonically e164 formatted. 

e.g. image

**Comment by moxie0** [comment]
in 3.20.0

**Pull request closed without merging**